streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -375,150 +199,117 @@ private RuntimeException unAssignChangeLogPartitions() {
-                          task.id(),
-                          t);
-                if (firstException == null) {
-                    firstException = t;
-                }
-            }
-        }
-
-        return firstException;
-    }
-
-
-
     void shutdown(final boolean clean) {
         log.debug("{} Shutting down all active tasks {}, standby tasks {}, suspended tasks {}, and suspended standby tasks {}",
-                  logPrefix, activeTasks.keySet(), standbyTasks.keySet(),
-                  suspendedTasks.keySet(), suspendedStandbyTasks.keySet());
-
-        for (final Task task : allTasks()) {
-            try {
-                task.close(clean);
-            } catch (final RuntimeException e) {
-                log.error("{} Failed while closing {} {} due to the following error:",
-                          logPrefix,
-                          task.getClass().getSimpleName(),
-                          task.id(),
-                          e);
-            }
-        }
+                  logPrefix, active.runningTaskIds(), standby.runningTaskIds(),
+                  active.previousTaskIds(), standby.previousTaskIds());
+
+        active.close(clean);
+        standby.close(clean);
         try {
             threadMetadataProvider.close();
         } catch (final Throwable e) {
             log.error("{} Failed to close KafkaStreamClient due to the following error:", logPrefix, e);
         }
         // remove the changelog partitions from restore consumer
         unAssignChangeLogPartitions();
+        taskCreator.close();
 
     }
 
     Set<TaskId> suspendedActiveTaskIds() {
-        return Collections.unmodifiableSet(suspendedTasks.keySet());
+        return active.previousTaskIds();
     }
 
     Set<TaskId> suspendedStandbyTaskIds() {
-        return Collections.unmodifiableSet(suspendedStandbyTasks.keySet());
+        return standby.previousTaskIds();
     }
 
-    void removeTasks() {
-        removeStreamTasks();
-        removeStandbyTasks();
+    Task activeTask(final TopicPartition partition) {
+        return active.runningTaskFor(partition);
     }
 
-    private void removeStandbyTasks() {
-        log.debug("{} Removing all standby tasks {}", logPrefix, standbyTasks.keySet());
-        standbyTasks.clear();
-        standbyTasksByPartition.clear();
+
+    Task standbyTask(final TopicPartition partition) {
+        return standby.runningTaskFor(partition);
     }
 
-    Task activeTask(final TopicPartition partition) {
-        return activeTasksByPartition.get(partition);
+    Map<TaskId, Task> activeTasks() {
+        return active.runningTaskMap();
     }
 
-    boolean hasStandbyTasks() {
-        return !standbyTasks.isEmpty();
+    void setConsumer(final Consumer<byte[], byte[]> consumer) {
+        this.consumer = consumer;
     }
 
-    Task standbyTask(final TopicPartition partition) {
-        return standbyTasksByPartition.get(partition);
+
+    boolean updateNewAndRestoringTasks() {
```

Comment:
Hmm... this function will always returns true, causing us to transit to `RUNNING` immediately?